gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/DagActionStore.java

```diff
@@ -51,30 +52,53 @@ public FlowId getFlowId() {
      */
     public DagAction updateFlowExecutionId(long eventTimeMillis) {
       return new DagAction(this.getFlowGroup(), this.getFlowName(),
-          String.valueOf(eventTimeMillis), this.getFlowActionType());
+          String.valueOf(eventTimeMillis), this.getJobName(), this.getDagActionType());
     }
   }
 
 
   /**
-   * Check if an action exists in dagAction store by flow group, flow name and flow execution id.
+   * Check if an action exists in dagAction store by flow group, flow name, flow execution id, and job name.
    * @param flowGroup flow group for the dag action
    * @param flowName flow name for the dag action
    * @param flowExecutionId flow execution for the dag action
-   * @param flowActionType the value of the dag action
+   * @param jobName job name for the dag action
```

Comment:
What's the motivation behind adding jobName to the DagAction? It's derived from the other 3 fields here, along with the flow edge and a random hash that is consistent on compilation. Dags should be uniquely identified by the other 3 fields right?